### PR TITLE
Revert "Remove pre commit"

### LINKED ISSUE #186: Re-enable pre-commit
It looks like branch protection rules and pre-commit (https://github.com/CDCgov/cfa_azure/pull/164/commits/60e1c870af996703523d65732925716600b4653e) were disabled in this repo. We need to turn them back on to enforce code review and prevent secrets from leaking.

### PR DESCRIPTION
**Pull request by kgostic**
This reverts commit 60e1c870af996703523d65732925716600b4653e to reinstate pre-commit in CI. Closes #186.

### CODE REVIEW

**Comment by kgostic** [comment]
Hang on, this is going to require some file cleanup.

**Comment by kgostic** [comment]
Once this is approved and merged, I'll add pre-commit to the branch protection ruleset.

File: .github/workflows/pre-commit.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
```
Comment (amondal2) [inline .github/workflows/pre-commit.yaml:6]: Looks like the default branch is called `master` in this repo, not `main`

Comment (kgostic) [reply .github/workflows/pre-commit.yaml]: Indeed. df884ab

**Review by amondal2 — APPROVED** [review]

**Review by natemcintosh — APPROVED** [review]

**Pull request merged**